amida-tech/amida-auth-microservice
# token verification and Bearer auth

#### What's this PR do?
Allows JWT-based identity verification, password updates, and `Bearer` auth for other services

#### Related JIRA tickets:
SER-6

#### How should this be manually tested?
Run the tests with `yarn test`. You can also manually create a user using the User `POST` endpoint, then use `/login` to get a JWT in return. This token should decode to contain the username provided. Here is my API session:

```HTTP
POST /api/user HTTP/1.1
Content-Type: application/json; charset=utf-8
Host: localhost:4000
Connection: close
User-Agent: Paw/3.1.3 (Macintosh; OS X/10.12.6) GCDHTTPRequest
Content-Length: 69

{"username":"test123","email":"test@amida.com","password":"testpass"}

POST /api/auth/login HTTP/1.1
Content-Type: application/json; charset=utf-8
Host: localhost:4000
Connection: close
User-Agent: Paw/3.1.3 (Macintosh; OS X/10.12.6) GCDHTTPRequest
Content-Length: 44

{"username":"test123","password":"testpass"}
```
